### DISCUSSION TITLE
best way to calculating time series mean&median value

### DISCUSSION THREAD

**Discussion by orttak** [corteva/rioxarray#337] [Q&A] [answered]
I try to get mean or median values of Sentinel COG time series over same tile. It took 3.5-4.5 min. I think my code block take too much time, when I compared gdal or qgis results. Is it right way? Any recommendation?

```
band_url1='https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/37/S/DA/2018/11/S2A_37SDA_20181109_0_L2A/B02.tif'
band_url2='https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/37/S/DA/2018/10/S2A_37SDA_20181030_0_L2A/B02.tif'

r38=rioxarray.open_rasterio(band_url1, masked=False, chunks=(1, "auto", -1))
r48=rioxarray.open_rasterio(band_url2, masked=False, chunks=(1, "auto", -1))
stack_bands2 = xr.concat([r38,r48], dim='band', join='outer' )
stack_bands2=stack_bands2.astype(np.float64)
stack_bands2.rio.write_nodata(np.nan, inplace=True)

#time info belongs to below code block
stack_bands2 = stack_bands2.mean(dim='band',skipna=True).compute()
#CPU times: user 6.28 s, sys: 2.52 s, total: 8.8 s
#Wall time: 3min 24s
```

**Comment by snowman2** [comment]
I would recommend asking that [here](https://github.com/pydata/xarray/discussions) as it is more of a xarray question and will likely get more expert advice about that. My recommendation is if you need more speed and are confident they cover the same location, you could try doing the operations on the dask arrays themselves instead of the xarray dataarrays as it may reduce some overhead.

**Reply by orttak** [reply]
Thanks. I asked xarray's page. If I'll get result, I'll share it

**Reply by RichardScottOZ** [reply]
Need the right environment type options set, too, for this to work more quickly.

e.g. https://github.com/rmg55/CloudDAAC_Binders/blob/main/s3_v_http.ipynb

**Comment by orttak** [comment, marked as answer]
From https://github.com/pydata/xarray/discussions/5330#discussioncomment-774852
" just checking: since the wall time is much bigger than the CPU time, did you try downloading those files before opening? I did have some trouble with downloads through rasterio some time ago."
